stock_picking_partner_info/views/stock_picking_views.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_picking_form" model="ir.ui.view">
+        <field name="name">stock.picking.form</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='partner_id']" position="attributes">
+                <attribute name="context">{'show_address': 1}</attribute>
+                <attribute name="options">{'always_reload': 1}</attribute>
+                <attribute
+                    name="attrs"
+                >{'invisible': [('picking_type_code', '!=', 'outgoing')]}</attribute>
+            </xpath>
```

Comment:
```suggestion
            <xpath expr="//field[@name='partner_id']" position="attributes">
                <attribute name="context">{'show_address': 1}</attribute>
                <attribute name="options">{'always_reload': 1}</attribute>
                <attribute
                    name="attrs"
                >{'invisible': [('picking_type_code', '!=', 'outgoing')]}</attribute>
            </xpath>
            <xpath expr="//field[@name='partner_id']" position="after">
                <field name="partner_id" attrs="{'invisible': [('picking_type_code', '=', 'outgoing')]}" />
            </xpath>
```
There was an oversight by me that the partner field was not showing for receipts, so we should either remove the invisible attribute or add another partner_id field to show in when the picking type code is not outgoing.